frontend/src/Connection.ts

```diff
@@ -408,8 +410,12 @@ export class Connection {
       }
     }
 
-    if (nodes.hasChangedSince(ref)) {
-      this.state = this.update({ nodes });
+    if (nodes.hasChangedSince(ref) && !this.updateThrottle) {
+      this.updateThrottle = true;
+      window.requestAnimationFrame(() => {
+        this.update({ nodes });
+        this.updateThrottle = false;
+      });
```

Comment:
> Nitpick: ref is not a great variable name, especially given it's defined way up high in the file. Can one rename local vars when destructuring in ts? sortedCollectionRef?

```javascript
const { ref: sortedCollectionRef } = nodes;
```

Reply:
> Question: would it be possible to check if we're updating a bit higher up in this method to avoid doing a bunch of work and then at the very end see "oh we're updating, never mind then"?

The updates to the collection still happen regardless of whether the `update` is called or not (they must to keep backend and frontend in sync, and I don't believe you could do this with purely immutable state with reasonable performance), so the work is never wasted. The final call to `update` just informs React that it needs to do a render.